OpenRA.Mods.Common/Activities/Parachute.cs

```diff
@@ -0,0 +1,77 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public class Parachute : Activity
+	{
+		WPos currentPosition;
+		WVec fallVector;
+
+		public Parachute(Actor self, WPos dropPosition)
+		{
+			var um = self.TraitOrDefault<UpgradeManager>();
+			if (um != null)
+				um.GrantUpgrade(self, "parachute", this);
+
+			var tpu = self.TraitOrDefault<TargetableParachutableUnit>();
+			if (tpu != null)
+				tpu.IsParachuting = true;
+
+			var p = self.Info.Traits.GetOrDefault<ParachutableInfo>();
+			if (p != null)
+				fallVector = new WVec(0, 0, p.FallRate);
```

Comment:
This is in `Activities/Parachute`, so it should require the `Parachutable` trait.
Otherwise we can have a `Freefall` activity for actors that don't have it :D